src/StoreApi/Routes/AbstractRoute.php

```diff
@@ -70,17 +70,69 @@ public function get_response( \WP_REST_Request $request ) {
 					$response = $this->get_route_response( $request );
 					break;
 			}
-			if ( 'GET' !== $request->get_method() && ! is_wp_error( $response ) ) {
-				$response->header( 'X-WC-Store-API-Nonce', wp_create_nonce( 'wc_store_api' ) );
-			}
 		} catch ( RouteException $error ) {
 			$response = $this->get_route_error_response( $error->getErrorCode(), $error->getMessage(), $error->getCode(), $error->getAdditionalData() );
 		} catch ( \Exception $error ) {
 			$response = $this->get_route_error_response( 'unknown_server_error', $error->getMessage(), 500 );
 		}
+
+		if ( is_wp_error( $response ) ) {
+			$response = $this->error_to_response( $response );
+		}
+
+		return $this->add_response_headers( $response );
+	}
+
+	/**
+	 * Add Store API headers to a response object.
+	 *
+	 * @param \WP_REST_Response $response Response object.
+	 * @return \WP_REST_Response
+	 */
+	protected function add_response_headers( $response ) {
```

Comment:
As an alternative, if the intention is to split up and group logic, but not abstract this, just make it private?

Reply:
Sure, but the naming is still ambiguous, it's only adding the store api nonce header at the moment. Currently this method is only called in one place and it could easily be replaced with just adding the header to the response in that one place. It seems like premature abstraction for something we might not even use to add additional headers to the response for.